mp/src/game/server/momentum/mom_triggers.cpp

```diff
@@ -1586,26 +1589,26 @@ void CNoGrenadesZone::OnEndTouch(CBaseEntity* pOther)
     pSticky->SetCanExplode(true);
 }
 
-bool CNoGrenadesZone::IsInsideNoGrenadesZone(CBaseEntity *pOther)
+CNoGrenadesZone* CNoGrenadesZone::IsInsideNoGrenadesZone(CBaseEntity *pOther)
 {
-    if ( pOther )
+    if (!pOther)
+        return NULL;
+
+    auto pSticky = dynamic_cast<CMomStickybomb *>(pOther);
+    FOR_EACH_VEC(s_vecNoGrenadeZones, i)
     {
-        const auto pSticky = dynamic_cast<CMomStickybomb*>(pOther);
-        FOR_EACH_VEC(s_vecNoGrenadeZones, i)
-        {
-            const auto pNoGrenadeZone = s_vecNoGrenadeZones[i];
+        const auto pNoGrenadeZone = s_vecNoGrenadeZones[i];
 
-            if (pNoGrenadeZone->m_bDisabled || !pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin()))
-                continue;
+        if (pNoGrenadeZone->m_bDisabled || !pNoGrenadeZone->PointIsWithin(pOther->GetAbsOrigin()))
+            continue;
 
-            if (!pSticky)
-                return true; // This is a rocket in a nonade zone and should fizzle
+        if (!pSticky)
+            return pNoGrenadeZone; // This is a rocket in a nonade zone and should fizzle
 
-            return !pNoGrenadeZone->m_bAirborneOnly || !pSticky->DidHitWorld();
-        }
+        return pNoGrenadeZone;
```

Comment:
At this point in the function it returns the zone regardless of whether or not it's a sticky, so you can remove the `pSticky` dynamic cast and just return `pNoGrenadeZone` here.